### PR TITLE
Added PHP 8.4 Support

### PR DESCRIPTION
**Pull request by manishranjan-adobe**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### Description

Tell us about why this change is necessary:
- Added PHP 8.4 Support
  - TARGET THE CURRENT RELEASE BRANCH

- Are you adding documentation?
  - TARGET THE CURRENT RELEASE BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - No

- Are you fixing a BC Break?
  - No

- Are you adding something the library currently does not support?
  - Adding upcoming PHP 8.4 Support

- Are you refactoring code?
  - No

### CODE REVIEW

**Comment by manishranjan-adobe** [comment]
@froschdesign Can you check this PR? All the checks have been passed. It would be great if you could take a moment to review it when you have the time. I’ve made sure to follow the contribution guidelines, but I’d appreciate any feedback or suggestions for improvement.

Looking forward to your insights!

**Comment by manishranjan-adobe** [comment]
@gsteel Can we please pick this?

File: docs/book/intro.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ if a user correctly provides the first word, the second is likely correctly ente
 improve OCR software for digitizing books.
 
 <!-- markdownlint-disable-next-line header-increment -->
```
Comment (gsteel) [inline docs/book/intro.md]:
```suggestion
<!-- markdownlint-disable-next-line heading-increment -->
```

The name of this issue was changed in markdown lint

File: docs/book/intro.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ if a user correctly provides the first word, the second is likely correctly ente
...
-> ### Additional Requirements
+> ## Additional Requirements
```
Comment (gsteel) [inline docs/book/intro.md]:
```suggestion
> ### Additional Requirements
```

**Review by gsteel — CHANGES_REQUESTED** [review]
Thanks @manishranjan-adobe - Just one tweak to the markdown file

File: docs/book/intro.md (hunk 1 above)
Comment (manishranjan-adobe) [reply docs/book/intro.md]: Its done.

File: docs/book/intro.md (hunk 2 above)
Comment (manishranjan-adobe) [reply docs/book/intro.md]: @gsteel This will result in failure of QA Checks (README Linting)

Comment (froschdesign) [reply docs/book/intro.md]: @manishranjan-adobe 
No, the rule is disabled. See: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md001---heading-levels-should-only-increment-by-one-level-at-a-time

Comment (manishranjan-adobe) [reply docs/book/intro.md]: @froschdesign I have corrected the piece of the line. cheers

File: .laminas-ci.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
     "ignore_php_platform_requirements": {
-        "8.3": true
+        "8.3": true,
+        "8.4": true
```
Comment (samsonasik) [inline .laminas-ci.json]: can this only ignore php 8.4 ?

**Review by samsonasik — APPROVED** [review]

File: composer.json [hunk 4]
```diff
@@ -16,18 +16,18 @@
         "rss": "https://github.com/laminas/laminas-recaptcha/releases.atom"
     },
     "require": {
-        "php": "~8.0.0 || ~8.1.0 || ~8.2.0 || ~8.3.0",
+        "php": "~8.0.0 || ~8.1.0 || ~8.2.0 || ~8.3.0 || ~8.4.0",
```
Comment (samsonasik) [inline composer.json]: imo, I think it is time to remove php `~8.0.0` support, other components no longer have php `~8.0.0` :)

File: .laminas-ci.json
Comment (alexmerlin) [inline .laminas-ci.json:1]: Missing a new line at the end of the file.

**Comment by samsonasik** [comment]
composer.lock needs to be regenerated

**Comment by samsonasik** [comment]
composer.json config platform needs to be updated to 8.1.99

**Comment by manishranjan-adobe** [comment]
@samsonasik The php config platform update is not passing the CI/CD pipeline. Please advise

**Comment by samsonasik** [comment]
Read CI error https://github.com/laminas/laminas-recaptcha/actions/runs/11495703202/job/31995792006?pr=39#step:3:74

File: composer.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
         "rss": "https://github.com/laminas/laminas-recaptcha/releases.atom"
     },
     "require": {
-        "php": "~8.1.0 || ~8.2.0 || ~8.3.0 || ~8.4.0",
+        "php": "~8.1.99 || ~8.2.0 || ~8.3.0 || ~8.4.0",
```
Comment (samsonasik) [inline composer.json]: rollback this, update config platform instead under config > platform

Comment (samsonasik) [reply composer.json]: see section https://github.com/laminas/laminas-recaptcha/blob/6f5c05905f82b09a81faf90233f3eaf67ead891f/composer.json#L52-L54

Comment (manishranjan-adobe) [reply composer.json]: Thanks @samsonasik for pointing out the mistake.

Comment (manishranjan-adobe) [reply composer.json]: The issue is resolved. cheers

**Review by samsonasik — APPROVED** [review]

**Review by gsteel — APPROVED** [review]
Thanks @manishranjan-adobe 👍

**Pull request merged**